utils/check_content_style.py

```diff
@@ -0,0 +1,185 @@
+		if re.search(first, line):
+			text_index += 1
+		else:
+			break
+	if text_index == 0:
+		error_list.append((1, "incorrect copyright header"))
+		return error_list
+	for line in lines[text_index:]:
+		if re.search(second, line):
+			text_index += 1
+		else:
+			break
+	if len(lines) - text_index < len(third):
+		error_list.append((len(third) - 1, "incomplete copyright header"))
+	else:
+		for index, (line, standard) in enumerate(zip(lines[text_index:], third)):
+			standard = standard.strip()
+			if standard != line:
+				error_list.append((text_index + index + 1, "incorrect copyright header"))
+				break
+	return error_list
+
+
+# Checks whether the file contains any non-ASCII characters. Extended ASCII codes are not accepted.
+# Parameters:
+# lines: the lines of the file
+# Returns: a list of tuples representing the location and the description of the formatting issues
+def check_ascii(lines):
+	error_list = []
+	for index, line in enumerate(lines):
+		for char in line:
+			if ord(char) < 0 or ord(char) > 127:
+				error_list.append((index + 1, "files should be plain ASCII"))
+	return error_list
+
+
+# Performs the regex-based formatting checks.
+# Parameters:
+# lines: the lines of the file
+# Returns: a list of tuples representing the location and the description of the formatting issues
+def check_regexes(lines):
+	error_list = []
+	for index, line in enumerate(lines):
+		for regex, description in regexes.items():
+			if re.search(regex, line):
+				error_list.append((index + 1, description))
+	return error_list
+
+
+# Checks that each line is properly indented based on the indentation of surrounding lines.
+# Parameters:
+# lines: the lines of the file
+# Returns: a list of tuples representing the location and the description of the formatting issues
+def check_indents(lines):
+	error_list = []
+	previous_indent = 0
+	for index, line in enumerate(lines):
+		indent = count_indent(line)
+		# Checking non-empty lines
+		if line and not line.isspace():
+			if indent - previous_indent > 1:
+				error_list.append((index + 1, "do not add more than a single level of indentation per line"))
+			previous_indent = indent
+	return error_list
+
+
+# Counts the number of tabs this line of text is indented with.
+# Parameters:
+# text: the text to count indents in
+# Returns: the number of leading tabulators
+def count_indent(text):
+	count = 0
+	for char in text:
+		if char == '\t':
+			count += 1
+		else:
+			break
+	return count
+
+
+# Generates and stores an error message. The error is later displayed in check_content_style().
+def print_error(file, line, reason):
+	error_list.append((line, "Formatting error in " + file + " line " + str(line) + ": " + reason))
+
+
+if __name__ == '__main__':
+	error_count = check_content_style()
+	if error_count > 0:
+		text = "Found " + str(error_count) + " formatting " + ("issue" if error_count == 1 else "issues") + "."
+		print(text)
+		exit(1)
+	else:
+		print("No formatting issues found.")
+	exit(0)
```

Comment:
The only place any file reads or console writes should occur is in this `'__main__'` block. All other methods should be taking a list of lines and returning a list of errors.